### PR TITLE
Add config to build dist package (emitted JS and .d.ts types)

### PR DESCRIPTION
**Pull request by jasongwartz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +18,7 @@
     "typecheck": "tsc --noEmit",
     "lint": "eslint .",
     "lint:fix": "eslint . --fix",
-    "build": "tsc --declaration",
+    "build": "tsc -p tsconfig.dist.json && mkdir -p dist/codegen && cp -r codegen/src dist/codegen/src",
```
Comment (jackkleeman) [inline package.json:21]: is the cp required - can we just tell tsconfig to emit there? and maybe it will make the dir for us idk

**Review by jackkleeman — APPROVED** [review]
approved with question

File: package.json (hunk 1 above)
Comment (jasongwartz) [reply package.json:21]: The `cp` is about copying the codegen-related `.pkl` files used by `pkl-gen-typescript`.

We could also use eg. https://www.npmjs.com/package/copyfiles if we didn't want to rely on `mkdir` and `cp` from the host.

Comment (jackkleeman) [reply package.json:21]: oh right, im dumb. looks good then :)

**Pull request merged**